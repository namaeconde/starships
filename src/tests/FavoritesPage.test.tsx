import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import { render, cleanup } from '@testing-library/react'
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import FavoritePage from '../pages/FavoritesPage';

const initialState = { favoriteList: { value: [] } }
const mockStore = configureStore()
let store

afterEach(cleanup);

test('FavoritesPage renders', () => {
  store = mockStore(initialState);
  const { asFragment } = render(<Provider store={store}><Router><FavoritePage /></Router></Provider>,);
  expect(asFragment()).toMatchSnapshot();
});
