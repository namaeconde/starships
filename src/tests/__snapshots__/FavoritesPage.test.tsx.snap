// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoritesPage renders 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-8atqhb"
  >
    <div
      class="MuiBox-root css-l10m55"
    >
      <div
        class="MuiBox-root css-175rvmj"
      >
        <a
          href="/"
        >
          <div
            class="MuiAvatar-root MuiAvatar-square css-ivlicw-MuiAvatar-root"
          >
            <img
              alt="Starship"
              class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
              src="/assets/home_logo.png"
            />
          </div>
        </a>
      </div>
      <div
        class="makeStyles-container-3 MuiBox-root css-vq34d8"
      />
      <div
        class="makeStyles-main-4 MuiBox-root css-1xkyxdb"
      >
        <div
          class="MuiBox-root css-uj0oyz"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-15ixj63-MuiTypography-root"
          >
            Favorites
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-undefined MuiBox-root css-1vtca5r-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiBox-root css-es3gb9-MuiGrid-root"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              You have no favorites.
            </p>
          </div>
        </div>
      </div>
      <div
        class="makeStyles-container-3 MuiBox-root css-1icjmr2"
      />
    </div>
  </div>
</DocumentFragment>
`;
